core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -302,8 +336,7 @@ public void transportTerminated() {
           log.warning("transportTerminated called without previous transportShutdown");
           activeTransport = null;
         }
-        // TODO(notcarl): replace this with something more meaningful
-        transportShutdown(Status.UNKNOWN);
+        clearActiveTransport();
```

Comment:
Good point.  Strangely transportShutdown() was being called here originally, which was in the same situation.